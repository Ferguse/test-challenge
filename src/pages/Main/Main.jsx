import { useEffect, useMemo } from 'react';
import { useDispatch } from 'react-redux';

import { Carousel } from 'components/Carousel';
import { CATEGORIES } from 'constants/Categories';
import { fetchCategories } from 'actions';

export const Main = () => {
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(fetchCategories(CATEGORIES));
  }, []);

  const carousels = useMemo(() => (
    CATEGORIES.map((category) => <Carousel key={category.name} category={category}/>)), []);

  return (
    <main className="content bg-white">
      <div className="container">
        {carousels}
      </div>
    </main>
  );
};


