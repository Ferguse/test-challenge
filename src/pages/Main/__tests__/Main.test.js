import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import configureMockedStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { createMemoryHistory } from 'history';
import { Router } from 'react-router-dom';

import { Main } from '../Main';

describe('Page [Main]', () => {
  const state = {
    wishList: {},
    categories: {
      entities: {
        bike: {
          test: {
            id: 'test',
            volumeInfo: {},
            category: {},
          },
        }},
    },
    current: {
      category: 'bike',
      book: 'test',
    },
  };

  const store = configureMockedStore([thunk])(state);
  const history = createMemoryHistory({ initialEntries: ['/'] });

  it('Renders', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router history={history} location={history.location} navigator={history}>
            <Main />
          </Router>
        </Provider>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});
