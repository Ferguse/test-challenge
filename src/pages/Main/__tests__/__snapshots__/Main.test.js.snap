// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page [Main] Renders 1`] = `
<main
  className="content bg-white"
>
  <div
    className="container"
  >
    <div
      className="carousel"
    >
      <h2
        className="title"
      >
        <a
          href="/category/bike"
          onClick={[Function]}
        >
          bike
        </a>
        <i
          className="carousel_indicator"
          style={
            {
              "backgroundColor": "#ffa2a2",
            }
          }
        />
      </h2>
      <div
        className="slick-slider slick-initialized"
      >
        
        <div
          className="slick-list"
        >
          <div
            className="slick-track"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            style={
              {
                "WebkitTransform": "translate3d(0px, 0px, 0px)",
                "WebkitTransition": "",
                "msTransform": "translateX(0px)",
                "opacity": 1,
                "transform": "translate3d(0px, 0px, 0px)",
                "transition": "",
              }
            }
          >
            <div
              aria-hidden={false}
              className="slick-slide slick-active slick-current"
              data-index={0}
              onClick={[Function]}
              style={
                {
                  "outline": "none",
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card"
                  style={
                    {
                      "--category-color": "#ffa2a2",
                    }
                  }
                >
                  <a
                    href="/category/bike/book/test"
                    onClick={[Function]}
                  >
                    <div
                      className="card-content flex direction-column justify-between"
                    >
                      <div
                        className="card-image flex justify-center"
                      >
                        <img
                          loading="lazy"
                        />
                        <i
                          className="fi fi-rs-books card-icon"
                        />
                      </div>
                      <div
                        className="card-info flex direction-column"
                      >
                        <span
                          className="card-title bold"
                        >
                          
                        </span>
                        <span>
                          NaN
                        </span>
                      </div>
                    </div>
                  </a>
                </div>
              </div>
            </div>
            <div
              aria-hidden={false}
              className="slick-slide slick-active slick-cloned"
              data-index={1}
              onClick={[Function]}
              style={
                {
                  "width": 0,
                }
              }
              tabIndex="-1"
            >
              <div>
                <div
                  className="card"
                  style={
                    {
                      "--category-color": "#ffa2a2",
                    }
                  }
                >
                  <a
                    href="/category/bike/book/test"
                    onClick={[Function]}
                  >
                    <div
                      className="card-content flex direction-column justify-between"
                    >
                      <div
                        className="card-image flex justify-center"
                      >
                        <img
                          loading="lazy"
                        />
                        <i
                          className="fi fi-rs-books card-icon"
                        />
                      </div>
                      <div
                        className="card-info flex direction-column"
                      >
                        <span
                          className="card-title bold"
                        >
                          
                        </span>
                        <span>
                          NaN
                        </span>
                      </div>
                    </div>
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        
        
      </div>
    </div>
    <div
      className="carousel"
    >
      <h2
        className="title"
      >
        <a
          href="/category/hiking"
          onClick={[Function]}
        >
          hiking
        </a>
        <i
          className="carousel_indicator"
          style={
            {
              "backgroundColor": "#a7fda7",
            }
          }
        />
      </h2>
      <div
        className="slick-slider slick-initialized"
      >
        
        <div
          className="slick-list"
        >
          <div
            className="slick-track"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            style={
              {
                "WebkitTransform": "translate3d(0px, 0px, 0px)",
                "WebkitTransition": "",
                "msTransform": "translateX(0px)",
                "opacity": 1,
                "transform": "translate3d(0px, 0px, 0px)",
                "transition": "",
              }
            }
          />
        </div>
        
        
      </div>
    </div>
    <div
      className="carousel"
    >
      <h2
        className="title"
      >
        <a
          href="/category/skating"
          onClick={[Function]}
        >
          skating
        </a>
        <i
          className="carousel_indicator"
          style={
            {
              "backgroundColor": "#a9a9fb",
            }
          }
        />
      </h2>
      <div
        className="slick-slider slick-initialized"
      >
        
        <div
          className="slick-list"
        >
          <div
            className="slick-track"
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
            onMouseOver={[Function]}
            style={
              {
                "WebkitTransform": "translate3d(0px, 0px, 0px)",
                "WebkitTransition": "",
                "msTransform": "translateX(0px)",
                "opacity": 1,
                "transform": "translate3d(0px, 0px, 0px)",
                "transition": "",
              }
            }
          />
        </div>
        
        
      </div>
    </div>
  </div>
</main>
`;
